src/Exercise/StringifyToDemystify.php

```diff
@@ -0,0 +1,157 @@
+    SelfCheck
+{
+    public function __construct(private Parser $parser)
+    {
+    }
+
+    public function getName(): string
+    {
+        return 'Stringify to Demystify';
+    }
+
+    public function getDescription(): string
+    {
+        return 'PHP 8\'s Stringable Interface';
+    }
+
+    public function getType(): ExerciseType
+    {
+        return ExerciseType::CGI();
+    }
+
+    public function configure(ExerciseDispatcher $dispatcher): void
+    {
+        $dispatcher->requireCheck(FunctionRequirementsCheck::class);
+    }
+
+    /**
+     * @return array<RequestInterface>
+     */
+    public function getRequests(): array
+    {
+        return array_map(
+            fn () => new Request(
+                'POST',
+                'https://phpschool.io/api',
+                [],
+                json_encode($this->getRandRequest(), JSON_THROW_ON_ERROR)
+            ),
+            array_fill(0, random_int(3, 6), null)
+        );
+    }
+
+    /**
+     * @return array{'success': bool, 'status': int, 'error'?: string, 'body'?: string}
+     * @throws \Exception
+     */
+    public function getRandRequest(): array
+    {
+        return [
+            [
+                'success' => true,
+                'status' => 200,
+                'body' => ['{fake: "data"}', '{stub: "payload"}'][random_int(0, 1)]
+            ],
+            [
+                'success' => false,
+                'status' => 400,
+                'error' => ['Bad Request: Incorrect mime type', 'Bad Request: Invalid key "id"'][random_int(0, 1)]
+            ],
+            [
+                'success' => false,
+                'status' => 401,
+                'error' => ['Unauthorized: Not authorised for this resource', 'Bad API token'][random_int(0, 1)]
+            ],
+            [
+                'success' => false,
+                'status' => 403,
+                'error' => ['Forbidden: Access forbidden', 'Forbidden: ACL not granted for resource'][random_int(0, 1)]
+            ],
+            [
+                'success' => false,
+                'status' => 500,
+                'error' => ['Server Error: Contact webmaster', 'Server Error: err_code: 3289327'][random_int(0, 1)]
+            ],
+        ][random_int(0, 4)];
+    }
+
+    public function getPatch(): Patch
+    {
+        $code = <<<CODE
+        function log_failure(\Stringable \$error) {
+            echo \$error;
+        }
+        CODE;
+
+        return (new Patch())->withInsertion(new CodeInsertion(CodeInsertion::TYPE_BEFORE, $code));
+    }
+
+    public function getRequiredFunctions(): array
+    {
+        return ['log_failure'];
+    }
+
+    public function getBannedFunctions(): array
+    {
+        return [];
+    }
+
+    public function check(Input $input): ResultInterface
+    {
+        $statements = $this->parser->parse((string) file_get_contents($input->getRequiredArgument('program')));
+
+        if (null === $statements) {
```

Comment:
We don't need to do this anymore: https://github.com/php-school/php-workshop/blob/master/src/Check/CodeExistsCheck.php

Reply:
this was to satisfy phpstan lol, can just doc block instead cause I guess we know for sure we won't be running this by if it's `null`

Reply:
yeah exactly, can just use `/** @var array<Stmt> $statements */` like I do in the others